phonopy/harmonic/force_constants.py

```diff
@@ -894,3 +935,45 @@ def permutation_error():
             permutation_error()
 
     return permutation
+
+# This can be thought of as computing 'map_atom_disp' and 'map_sym' for all atoms,
+# except done using permutations instead of by computing overlaps.
+#
+# Input:
+#  * permutations, shape [num_rot][num_pos]
+#  * atom_list_done
+#
+# Output:
+#  * map_atoms, shape [num_pos].
+#    Maps each atom in the full structure to its equivalent atom in atom_list_done.
+#    (each entry will be an integer found in atom_list_done)
+#
+#  * map_syms, shape [num_pos].
+#    For each atom, provides the index of a rotation that maps it into atom_list_done.
+#    (there might be more than one such rotation, but only one will be returned)
+#    (each entry will be an integer 0 <= i < num_rot)
+def _get_sym_mappings_from_permutations(permutations,
+                                        atom_list_done):
```

Comment:
this function probably isn't necessary.  I added it because I had big plans on how it could be optimized, but after benchmarking I found that it is plenty fast already, even as a brute force loop!

I don't know how much faster it is than e.g. get_atom_mapping_by_symmetry, but I doubt it is by very much.